pkg/proxy/iptables/proxier.go

```diff
@@ -1030,6 +1030,15 @@ func (proxier *Proxier) syncProxyRules() {
 			hasEndpoints = len(allEndpoints) > 0
 		}
 
+		// Service InternalTrafficPolicy is only enabled when all of the
+		// following are true:
+		// 1. InternalTrafficPolicy is PreferLocal or Local
+		// 2. ServiceInternalTrafficPolicy feature gate is on
+		if svcInfo.PreferNodeLocalEndpoints() && utilfeature.DefaultFeatureGate.Enabled(features.ServiceInternalTrafficPolicy) {
```

Comment:
Can we add a unit test for the iptable rules? I don't think we have to test the iptables datapath (that would be e2e tests) but testing the expected rules is probably good. See `TestEndpointSliceE2E` for an example.

Reply:
done